CMakeLists.txt

```diff
@@ -11,8 +11,35 @@ include(ExternalProject)
 include(FindPkgConfig)
 include(GNUInstallDirs)
 
+# Build Options
+option(USE_Z5 "Support OMEZarr" OFF)
+option(USE_DCMTK "Use DCMTK" OFF)
+option(USE_ARROW "Use Arrow" OFF)
+option(BUILD_WITH_ALL "Building with all I/O support" OFF)
+option(MINIMUM_BUILD "Building with Only Tiff and Pybind11(For Python package)" ON)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake-modules")
+
+if(MINIMUM_BUILD)
+	set(USE_Z5 OFF)
+	set(USE_DCMTK OFF)
+	set(USE_ARROW OFF) 
+endif()
+
+if(BUILD_WITH_ALL)
```

Comment:
Great PR overall. The only thing that I'd change/improve is the naming "MINIMUM_BUILD - BUILD_WITH_ALL". To me, "BUILD_WITH_ALL" sounds rather non-intuitive and not self-explanatory (Build with all the <what>?) Secondly, the pair "MINIMUM_BUILD - BUILD_WITH_ALL" is non-symmetric morphologically. Thirdly, I would stick to underscore-free flag convention as it reduces guessing whether words are separated or not. The naming natural to me would be "MINBUILD - MAXBUILD" or "NOEXTRAS - ALLEXTRAS" or so, but it's only my subjective suggestion.